### PR TITLE
New getAttributes method for complex set of objects

### PR DESCRIPTION
**Pull request by stavamichal**
- method expects member, group and resource objects (with some
 connection between them) and list of attrNames
 - method is able to return any attribute from namespaces: user,
 user-facility, facility, member, resource, member-resource, group,
 member-group and group-resource
 - test is also part of this commit
 - method was added to RPC as additional version of getAttributes

### CODE REVIEW

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AttributesManagerMethod.java [hunk 1]
```diff
@@ -105,6 +105,30 @@
...
+	 * - resource
+	 * - member-resource
+	 * - group-resource
+	 * - user (get from member object)
+	 * - facility (get from resource object)
+	 * - user-facility
+	 *
+	 * @param member int Member <code>id</code>
+	 * @param group int Group <code>id</code>
+	 * @param resource in Resource <code>id</code>
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AttributesManagerMethod.java]: in -> int

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AttributesManagerMethod.java [hunk 2]
```diff
@@ -105,6 +105,30 @@
...
+	 * @param attrNames List<String> Attribute names
+	 * @return List<Attribute> All attributes from supported namespaces.
+	 * @throw MemberNotExistsException When Member with <code>id</code> doesn't exist.
+	 * @throw ResourceNotExistsException When Resource with <code>id</code> doesn't exist.
+	 * @throw GroupNotExistsException When Group with <code>id</code> doesn't exist.
+	 * @throw MemberResourceMismatchException When Member is not from the same Vo as Resource.
+	 * @throw GroupResourceMismatchException When Group is not from the same Vo as Resource.
+	 * @throws InternalErrorException When any internal error occurs.
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AttributesManagerMethod.java]: This exception is added by default by our docs generator.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 3]
```diff
@@ -457,6 +457,58 @@ public AttributesManagerBlImpl(AttributesManagerImplApi attributesManagerImpl) {
...
+		}
+
+		//return all group and group_resource attributes
+		if(!groupAndGroupResourceAttrNames.isEmpty()) attributes.addAll(this.getAttributes(sess, resource, group, groupAndGroupResourceAttrNames, true));
+		//return all member_group attributes
+		if(!memberGroupAttrNames.isEmpty()) attributes.addAll(this.getAttributes(sess, member, group, memberGroupAttrNames, false));
+		//return all user, member, member-resource and user-facility attributes
+		if(!memberResourceAttrNames.isEmpty()) attributes.addAll(this.getAttributes(sess, member, resource, memberResourceAttrNames, true));
+		//return all resource attributes
+		if(!facilityAttrNames.isEmpty()) attributes.addAll(this.getAttributes(sess, resource, resourceAttrNames));
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: Should be checked on ``resourceAttrNames`` and not ``facilityAttrNames``.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 4]
```diff
@@ -457,6 +457,58 @@ public AttributesManagerBlImpl(AttributesManagerImplApi attributesManagerImpl) {
...
+		//return all facility attributes
+		if(!resourceAttrNames.isEmpty()) attributes.addAll(this.getAttributes(sess, facility, facilityAttrNames));
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: Should be checked on ``facilityAttrNames`` and not ``resourceAttrNames``.

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AttributesManagerMethod.java (hunk 1 above)
Comment (stavamichal) [reply perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AttributesManagerMethod.java]: Fixed.

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AttributesManagerMethod.java (hunk 2 above)
Comment (stavamichal) [reply perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AttributesManagerMethod.java]: Removed.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java (hunk 3 above)
Comment (stavamichal) [reply perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: Fixed. Thank you for check.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java (hunk 4 above)
Comment (stavamichal) [reply perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: Fixed. Thank you for check.

**Comment by stavamichal** [comment]
Should be fixed now, please check again.

**Pull request merged**